### ISSUE TITLE
lines-after-imports is not being enforced correctly on 0.1.3

### ISSUE THREAD

**Issue by timbedard** [astral-sh/ruff#8270]
Ruff version: 0.1.3

ruff.toml:
```toml
[isort]
lines-after-imports = 2
```

Python file:
```python
import os
print(os.environ["SHELL"])
```

Expected:
```python
import os


print(os.environ["SHELL"])
```

Actual:
```python
import os

print(os.environ["SHELL"])
```

In fact, it seems that the lines after the imports remain untouched if there are 1 or 2 lines and are only altered from 0 -> 1 and >2 -> 2.

With `lines-after-imports = 1`, it should always be 1 line.

With `lines-after-imports = 2`, it should always be 2 lines.

🙏

**Comment by dhruvmanila** [comment]
Hey, I'm unable to reproduce this locally. Can you tell us how are you running Ruff? Are you using the CLI / VSCode extension / any other editor with ruff-lsp?

**Comment by timbedard** [comment]
> Hey, I'm unable to reproduce this locally. Can you tell us how are you running Ruff? Are you using the CLI / VSCode extension / any other editor with ruff-lsp?

Via CLI.

I think I've figured it out though. The lines are only adjusted when running `ruff --fix`, not when running `ruff format`. That's a little surprising.

First thought, best thought: Shouldn't running `ruff format` also rope in the isort functionality? I understand the history of how we got here (integrating isort functionality before `ruff format` was around), but the end of the day, isort is primarily a formatter as well, so it's bit odd that only black functionality is tied to `ruff format` while isort is relegated to `ruff --fix`.

Thoughts?

**Comment by MichaReiser** [comment]
We agree that `isort` is the weird one know and we are considering pulling it out of linting. Whether it will be part of formatting or entirely its own command, we don't know yet.

**Comment by timbedard** [comment]
> We agree that `isort` is the weird one now and we are considering pulling it out of linting. Whether it will be part of formatting or entirely its own command, we don't know yet.

Cool, cool. I'll close this issue and keep an eye out for changes down the line.

**Comment by yutotakano** [comment]
Could I request this to be re-opened?

Setting `lines-after-import` still doesn't enforce the number of blank lines if it's already 1 or 2, it seems. 
(https://play.ruff.rs/4529c0ac-4c6f-45e0-9c3f-fc4cb29f0e9a?secondary=Format)

Black's 2026 stable style added `always_one_newline_after_import` which always changes 2 blank lines to 1 line. I see that the Ruff 2026 style guide also mentioned this, but I think it was closed without resolution: https://github.com/astral-sh/ruff/issues/20482
This issue we're on right now is probably the right one to re-open in order to address that.

It just feels weird to me that the formatter is supposed to be rather uncompromising (if it followed Black's steps) but that I could have either 1 or 2 blank lines after imports regardless of what I set as `lines-after-import`.

**Comment by ntBre** [comment]
@yutotakano There's a newer issue tracking this in https://github.com/astral-sh/ruff/issues/17558, so I don't think we need to reopen this issue.

As noted above, this setting is primarily for the isort rules in the _linter_ not the formatter, so the setting is enforced when running something like:

```shell
ruff check --select I001 --fix
```

rather than

```shell
ruff format
```

However, there's still an incompatibility between the isort rule and the formatter for some values of `lines-after-imports`, which is what's tracked in #17558.
